Equilobe.DailyReport.SL/ReportService.cs

```diff
@@ -12,27 +13,97 @@ public class ReportService : IReportService
     {
         public IJiraService JiraService { get; set; }
 
+        public JiraRequestContext JiraRequestContext { get; set; }
+
         #region IReportService Implementation
         public void UpdateDashboardData(long instanceId)
         {
-            UpdateAtlassianUsers(instanceId);
+            JiraRequestContext = GetJiraRequestContext(instanceId);
+
+            SyncAtlassianUsers(instanceId);
+            SyncAtlassianWorklogs(instanceId);
         }
         #endregion
 
         #region Helpers
-        private void UpdateAtlassianUsers(long instanceId)
+        private void SyncAtlassianWorklogs(long instanceId)
         {
-            var jiraRequestContext = GetJiraRequestContext(instanceId);
-            var users = JiraService.GetAllUsers(jiraRequestContext)
-                .Select(p => ToAtlassianUser(p, instanceId))
-                .ToList();
+            SyncDeletedWorklogs(instanceId);
+            SyncUpdatedWorklogs(instanceId);
+        }
+
+        private void SyncDeletedWorklogs(long instanceId)
+        {
+            var deletedWorklogsIds = JiraService.GetDeletedWorklogsIds(JiraRequestContext, DateTime.UtcNow.AddMonths(-7));
+
+            if (deletedWorklogsIds == null)
+                return;
 
             using (var db = new ReportsDb())
             {
-                var dbUsers = db.AtlassianUser
+                var dbWorklogs = db.AtlassianWorklog
```

Comment:
You could refactor to something like:             
```
var dbWorklogs = db.AtlassianWorklog
                    .Where(w => w.InstalledInstanceId == instanceId)
                    .Where(w => deletedWorklogsIds.Contains(w.JiraWorklogId))
                    .ToList();

db.AtlassianWorklog.RemoveRange(dbWorklogs);
db.SaveChanges();
```


And you lose a little extra code

Reply:
Done.